packages/muelu/src/Misc/MueLu_MultiVectorAggregationTransferFactory_decl.hpp

```diff
@@ -0,0 +1,142 @@
+#define MUELU_MULTIVECTORAGGREGATIONTRANSFER_FACTORY_DECL_HPP
+
+#include "MueLu_ConfigDefs.hpp"
+#include "MueLu_TwoLevelFactoryBase.hpp"
+#include "Xpetra_MultiVector_fwd.hpp"
+#include "Xpetra_MultiVectorFactory_fwd.hpp"
+#include "Xpetra_Matrix.hpp"
+
+#include "MueLu_CoarseMapFactory_fwd.hpp"
+#include "MueLu_MultiVectorAggregationTransferFactory_fwd.hpp"
+#include "MueLu_Utilities_fwd.hpp"
+
+namespace MueLu {
+
+/*!
+  @class MultiVectorAggregationTransferFactory class.
+  @brief Class for transferring a MultiVector from a finer level to a coarser one using aggregates
+
+    This is separate from MultiVectorTransferFactory and CoordinatesTransferFactory
+    as it transfers non-coordinate quantities based only on aggregate information.
+
+  ## Input/output of MultiVectorAggregationTransferFactory ##
+
+  ### User parameters of MultiVectorAggregationTransferFactory ###
+  Parameter | type | default | master.xml | validated | requested | description
+  ----------|------|---------|:----------:|:---------:|:---------:|------------
+  | Vector name| string | "undefined" |   | * | (*) | Name of vector
+  | Vector factory | Factory | null |   | * | (*) | Factory providing vector
+
+  The * in the @c master.xml column denotes that the parameter is defined in the @c master.xml file.<br>
+  The * in the @c validated column means that the parameter is declared in the list of valid input parameters (see MultiVectorAggregationTransferFactory::GetValidParameters).<br>
+  The * in the @c requested column states that the data is requested as input with all dependencies (see MultiVectorAggregationTransferFactory::DeclareInput).
+
+  The factory first checks whether there is already valid coarse information available on the coarse level. 
+  If that is the case, we skip the transfer and reuse the available information.
+  Otherwise we try to build the coarse grid vector by using the information about the
+  aggregates, the fine level vector and the coarse map information.
+
+  ### Variables provided by MultiVectorAggregationTransferFactory ###
+
+  After MultiVectorAggregationTransferFactory::Build the following data is available (if requested)
+
+  Parameter | generated by | description
+  ----------|--------------|------------
+  | Vector name | MultiVectorAggregationTransferFactory   | coarse level vector
+*/
+  template<class Scalar = DefaultScalar,
+           class LocalOrdinal = DefaultLocalOrdinal,
+           class GlobalOrdinal = DefaultGlobalOrdinal,
+           class Node = DefaultNode>
+  class MultiVectorAggregationTransferFactory : public TwoLevelFactoryBase {
+#undef MUELU_MULTIVECTORAGGREGATIONTRANSFERFACTORY_SHORT
+#include "MueLu_UseShortNames.hpp"
+
+  public:
+    //! @name Constructors/Destructors.
+    //@{
+
+    //! Constructor.
+    MultiVectorAggregationTransferFactory() { }
+
+    //! Destructor.
+    virtual ~MultiVectorAggregationTransferFactory() { }
+
+    RCP<const ParameterList> GetValidParameterList() const;
```

Comment:
Technically, this is not a constructor/destructor. Hence, the doxygen grouping is not exact here.